src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -74,6 +75,18 @@ const ProgressiveProfiling = (props) => {
     }
   }, [DASHBOARD_URL, enablePersonalizedRecommendations, registrationResponse]);
 
+  // get optimizely variation for recommendations page exp
+  useEffect(() => {
+    let timer = null;
+    if (showRecommendationsPage) {
+      activateRecommendationsExperiment();
+      timer = setTimeout(() => {
+        setShowRecommendationsPage(isUserInVariation());
+      }, 500);
+    }
+    return () => clearTimeout(timer);
+  }, [showRecommendationsPage]);
```

Comment:
If the recommendations are already set to true and then we activate the experiment and find that the recommendations are not enabled for this user we will be switching between "Next"/"Submit" text. 

Should we not combine the conditions? 
Maybe move the complete recommendations logic in a separate useEffect?

Reply:
good catch. thanks